wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WBeanComponent.java

```diff
@@ -169,7 +169,12 @@ public Object getBeanId() {
 	 */
 	@Override
 	public void setBeanProperty(final String propertyName) {
-		getOrCreateComponentModel().setBeanProperty(propertyName);
+		String currBeanProperty = getBeanProperty();
+
+		if ((propertyName != null && currBeanProperty == null)
+			|| (currBeanProperty != null && (propertyName == null || !currBeanProperty.equals(propertyName)))) {
```

Comment:
This style of `if statement` could be simplified by using `!Objects.equals(currBeanProperty, propertyName)`. Replace this pattern that is used in other areas. It would be good to unit test each of these scenarios to make sure a new model is created or not created.